src/FakeItEasy/Creation/DummyValueResolver.cs

```diff
@@ -66,12 +65,12 @@ private bool TryResolveDummyValueWithAllAvailableStrategies(DummyCreationSession
             {
                 if (strategy.TryCreateDummyValue(session, typeOfDummy, out result))
                 {
-                    this.strategyCache.Add(typeOfDummy, strategy);
+                    this.strategyCache.TryAdd(typeOfDummy, strategy);

```

Comment:
> You mean if it throws?

No, I mean if it returns `false`. I'm not really worried about this, just making sure we had this case covered. I hadn't realized it would only return false if the key already existed.